### ISSUE TITLE
h1 line height different in Safari and Firefox (last version)

### ISSUE THREAD

**Issue by jeremenichelli** [necolas/normalize.css#593]
I've noticed a difference in the line height of  `<h1>` tags between these two and I've created this issue to know if normalize tries to mitigate this kind of differences too.

In case maintainers feel it should I can take care of creating the pull request. Thank you.

**Comment by Karklis** [comment]
if you zoon out on Firefox to 80% show the same size as in Safari or Chrome?

**Comment by jeremenichelli** [comment]
@Karklis nope, in Safari and Chrome the `line-height` is `50px` and in Firefox it's `39.4px' and zooming out barely increases it to `39.9px`

**Comment by Karklis** [comment]
Could you provide code and all steps to reproduce this behavior? Are you using just default normalize css without any other css code, or mix something?

**Comment by jeremenichelli** [comment]
Create a new page with this content:

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Line height diff in h1 tags</title>

    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/normalize/4.1.1/normalize.min.css">
</head>
<body>
    <h1>line height diff in h1 tags</h1>

    <p>line-height: <span id="lh"></span></p>
    <script>
        document.querySelector('#lh').textContent = window.getComputedStyle(document.querySelector('h1'))[ 'lineHeight' ];
    </script>
</body>
</html>
```

You will be able to see the computed line height on page or open the web inspector in Safari and Firefox and check the resulting computed styles for `line-height` property.

### Results

**Firefox:** `33px`
**Safari:** `37px`

Considering that in both the `font-size` is `32px`, conclusions point that Firefox is assigning only `1.03em` of `line-height` while Safari and Chrome `1.15625em`.

**Comment by jeremenichelli** [comment]
Hi @Karklis, were you able to test this? Do you prefer another presentation for the test case?

**Comment by jonathantneal** [comment]
I’ve put your test on CodePen. I see a difference between Chrome and Firefox, between Firefox Mac and Firefox PC, and possibly by a fraction of a pixel in Edge/IE.

http://codepen.io/jonneal/details/RRNpLB/

**Comment by jeremenichelli** [comment]
So, what do we take as a standard to make them all match? I'll create the
PR :)

On Tuesday, 31 May 2016, Jonathan Neal <notifications@github.com> wrote:

> I’ve put your test on CodePen. I see a difference between Chrome and
> Firefox, between Firefox Mac and Firefox PC, and possibly by a fraction of
> a pixel in Edge/IE.
>
> http://codepen.io/jonneal/details/RRNpLB/
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/necolas/normalize.css/issues/593#issuecomment-222878823>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADrYYaQwXDyO_wsmqmZymKdVHFAEyEooks5qHPNYgaJpZM4IWzOl>
> .
>


-- 

*Jeremias Menichelli*


*web developer*
skype | jeremias.menichelli
<http://jeremenichelli.github.io>
github | jeremenichelli

**Comment by jonathantneal** [comment]
The [CodePen test](http://codepen.io/jonneal/details/RRNpLB/) measures the `height` of single-line `h1` headings using either the initial or **sans-serif** font. Here are its [results](https://app.crossbrowsertesting.com/public/i76b092cd2b52b86/screenshots/z091b48369aa85af0b9b?size=small&type=windowed) when run with a large stack of popular browsers.

| Browser     | Font       | Line Height  |
| ----------- | ---------- | ------------ |
| And Chrome  | initial    | 136px        |
| And Chrome  | sans-serif | 117px        |
| Mac Chrome  | initial    | 115px        |
| Mac Chrome  | sans-serif | 115px        |
| Win Chrome  | initial    | 115px        |
| Win Chrome  | sans-serif | 115px        |
| Mac Firefox | initial    | 120px        |
| Mac Firefox | sans-serif | 101px        |
| Win Firefox | initial    | 118px        |
| Win Firefox | sans-serif | 118px        |
| iOS Safari  | initial    | 117px        |
| iOS Safari  | sans-serif | 117px        |
| Mac Safari  | initial    | 115px        |
| Mac Safari  | sans-serif | 115px        |
| Win Edge    | initial    | 114.999px    |
| Win Edge    | sans-serif | 114.999px    |
| Win IE11    | initial    | 114.999px    |
| Win IE11    | sans-serif | 114.999px    |
| Win IE10    | initial    | 114.999px    |
| Win IE10    | sans-serif | 114.999px    |
| Win IE9     | initial    | 114.999px    |
| Win IE9     | sans-serif | 114.999px    |
| Mac Opera   | initial    | 115px        |
| Mac Opera   | sans-serif | 115px        |
| Win Opera   | initial    | 115px        |
| Win Opera   | sans-serif | 115px        |

When the **font size** was **100px**, the most common **line height** was **115px**. However, results varied from **101px** on **Mac Firefox** to **136px** on **Android Chrome**.

**Comment by jeremenichelli** [comment]
So it seems like something between `1.5em` and `1.15625em` should be the _normalized_ value of the line-height. I saw you noticed differences in both `h1` and `h2` elements, should we applied this fix to all headings?

**Comment by jonathantneal** [comment]
And here are [the results](https://app.crossbrowsertesting.com/public/i76b092cd2b52b86/screenshots/zaf25b53e6c075ec2db2) for a [normalized test](http://s.codepen.io/jonneal/debug/YWPEQW) where the **line height** of **h1** headings is defined as **1.15**.

| Browser     | Font       | Line Height  |
| ----------- | ---------- | ------------ |
| And Chrome  | initial    | 115px        |
| And Chrome  | sans-serif | 115px        |
| Mac Chrome  | initial    | 115px        |
| Mac Chrome  | sans-serif | 115px        |
| Win Chrome  | initial    | 115px        |
| Win Chrome  | sans-serif | 115px        |
| Mac Firefox | initial    | 115px        |
| Mac Firefox | sans-serif | 115px        |
| Win Firefox | initial    | 115px        |
| Win Firefox | sans-serif | 115px        |
| iOS Safari  | initial    | 115px        |
| iOS Safari  | sans-serif | 115px        |
| Mac Safari  | initial    | 115px        |
| Mac Safari  | sans-serif | 115px        |
| Win Edge    | initial    | 115px        |
| Win Edge    | sans-serif | 115px        |
| Win IE11    | initial    | 115px        |
| Win IE11    | sans-serif | 115px        |
| Win IE10    | initial    | 115px        |
| Win IE10    | sans-serif | 115px        |
| Win IE9     | initial    | 115px        |
| Win IE9     | sans-serif | 115px        |
| Mac Opera   | initial    | 115px        |
| Mac Opera   | sans-serif | 115px        |
| Win Opera   | initial    | 115px        |
| Win Opera   | sans-serif | 115px        |

When the **font size** was **100px**, all **line height** was **115px** with no variation.

**Comment by jeremenichelli** [comment]
OK, so should I make the PR containing fix for all the heading tags to `1.15`?

**Comment by jonathantneal** [comment]
Sure. Remember, the comment will need to specify which browsers this normalization is targeting, and this would be the most particular list I have yet seen in the project, and you’ll need to manage the line breaks to follow the style guide of the project. If there are issues, I will still take in your PR, but also correct them during the merge.

**Comment by jeremenichelli** [comment]
Great, will get that done and tag you. Thank you for your help.

**Comment by jonathantneal** [comment]
Thanks :)

**Comment by klig1** [comment]
Digging this up as I've got another issue with line height on firefox. 
See codepen attached [here](https://codepen.io/clem1/pen/BaNEZWz)

Firefox: (text is not vertically aligned on the background color)
![image](https://user-images.githubusercontent.com/62945328/78259502-b40fd480-74f4-11ea-9fe2-897f6a4cfa9a.png)

Chrome: (text is aligned)
![image](https://user-images.githubusercontent.com/62945328/78259605-d3a6fd00-74f4-11ea-85a9-8b60f8a331b0.png)

Any idea on how to resolve this? Tried with the 1.15 fix but didn't help. 

Thanks
